src/extension.ts

```diff
@@ -134,6 +135,20 @@ export async function activate(context: vscode.ExtensionContext) {
 
         await ExtensionDisposableFiles.initialize(context)
 
+        // TODO: Completion providers may be registered per-workspace folder instead of session-wide.
+        //       Should we watch the open workspace folders, and dynamically register/deregister the
+        //       completion provider based on whether the workspace folder looks like it contains a
+        //       SAM app?
```

Comment:
There is one functional difference: If we registered against a workspace folder we could match against `.aws/parameters.json` rather than `**/.aws/parameters.json`. But I agree that this is premature optimization-comment removed. ✔️